src/deluge/gui/ui/rename/rename_clipname_ui.cpp

```diff
@@ -84,9 +84,10 @@ ActionResult RenameClipNameUI::buttonAction(deluge::hid::Button b, bool on, bool
 
 void RenameClipNameUI::enterKeyPress() {
 
-	// If actually changing it...
-	if (!clip->clipName.equalsCaseIrrespective(&enteredText)) {
-		if (currentSong->getAudioOutputFromName(&enteredText)) {
+	// Don't allow duplicate names on clips of a single output.
+	if (!clip->name.equalsCaseIrrespective(&enteredText)) {
+		Clip* other = currentSong->getClipFromName(&enteredText);
+		if (other && other->output == clip->output) {
```

Comment:
this is not a sufficient check, getClipFromName returns the first clip with the name even if it's a different output from the clip being renamed

Reply:
fixed in https://github.com/SynthstromAudible/DelugeFirmware/pull/3056